lib/pages/company_site/eti/header.rb

```diff
@@ -18,20 +19,32 @@ def search_product(name, params = {})
         self.search_string = name
         search_button
         wait_render_table
+      end
 
-        # Костыль для достоверности после sleep 3, который выполняет поиск до тех пор, пока не появится товар
-        # Событие создания товара может не успеть записаться в эластик => товар не выведется после первого поиска
+      def retry_search_product(name, params = {})
+        exact_search if params[:exact] == true
+        self.search_type = params[:search_type]
+        self.search_string = name
+        search_button
+        wait_render_table
+        # Повторный поиск до тех пор пока не появится нужное количество товаров
         table_products = CompanySite::ETI::Table::Products.new
         max_attempts = 10
-
-        max_attempts.times do
-          break unless table_products.string_no_products?
-
+        # Запускается цикл с определенным количеством повторов (10)
+        # Каждую итерацию заново жмём кнопку поиска пока скопированный товар не появится в идексе эластика
+        max_attempts.times do |attempt|
           search_button
           wait_render_table
+        # Если товары отобразились идём дальше
+          unless table_products.string_no_products?
+            # Если на странице два и более товара метод завершается, дальнейшие попытки поиска не нужны
+            return if table_products.products_elements.size >= 2
```

Comment:
Lint/NonLocalExitFromIterator: Non-local exit from iterator, without return value. next, break, Array#find, Array#any?, etc. is preferred.<br>[Correctable] Style/SoleNestedConditional: Consider merging nested conditions into outer unless conditions.